File: src/main/java/org/motivepick/config/FacebookConfig.java
package org.motivepick.config

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;

@Configuration
class FacebookConfig implements Oauth2Config {

    private String clientId;
    private String clientSecret;
    private String userAuthorizationUri;
    private String accessTokenUri;
    private String userInfoUri;


    @Override
    @Value("${facebook.clientId}")
    public String getClientId() {
        return clientId;
    }

    public void setClientId(String clientId) {
        this.clientId = clientId;
    }

    @Override
    @Value("${facebook.clientSecret}")
    public String getClientSecret() {
        return clientSecret;
    }

    public void setClientSecret(String clientSecret) {
        this.clientSecret = clientSecret;
    }

    @Override
    @Value("${facebook.userAuthorizationUri}")
    public String getUserAuthorizationUri() {
        return userAuthorizationUri;
    }

    public void setUserAuthorizationUri(String userAuthorizationUri) {
        this.userAuthorizationUri = userAuthorizationUri;
    }

    @Override
    @Value("${facebook.accessTokenUri}")
    public String getAccessTokenUri() {
        return accessTokenUri;
    }

    public void setAccessTokenUri(String accessTokenUri) {
        this.accessTokenUri = accessTokenUri;
    }

    @Override
    @Value("${facebook.userInfoUri}")
    public String getUserInfoUri() {
        return userInfoUri;
    }

    public void setUserInfoUri(String userInfoUri) {
        this.userInfoUri = userInfoUri;
    }
}

File: src/main/kotlin/org/motivepick/config/FacebookConfig.kt
package org.motivepick.config

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Configuration

@Configuration
class FacebookConfig (

    @Value("\${facebook.clientId}")
    override val clientId: String,

    @Value("\${facebook.clientSecret}")
    override val clientSecret: String,

    @Value("\${facebook.userAuthorizationUri}")
    override val userAuthorizationUri: String,

    @Value("\${facebook.accessTokenUri}")
    override val accessTokenUri: String,

    @Value("\${facebook.userInfoUri}")
    val userInfoUri: String
) : Oauth2Config